### ISSUE TITLE
develop 版本 nn.functional.softmax 在部分输入为 -inf 的特殊 case 上输出全为 NaN

### ISSUE THREAD

**Issue by LemonNoel** [PaddlePaddle/Paddle#51904]
### bug描述 Describe the Bug

PaddlePaddle 版本：develop（commit id：c658be9b6086d287e92214ba85d43de9261933d6)

在个别包括 INF 的 case 会出现输出全部 NaN 的情况，在相同case 上， 2.4.2 版本结果符合预期。

```
# develop 版本
>>> paddle.nn.functional.softmax(x, axis=-1)
Tensor(shape=[2, 250080], dtype=float32, place=Place(gpu:0), stop_gradient=True,
       [[nan, nan, nan, ..., nan, nan, nan],
        [nan, nan, nan, ..., nan, nan, nan]])

# 2.4.2 版本
>>> paddle.nn.functional.softmax(x, axis=-1)
W0321 13:57:01.614192 73276 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 7.0, Driver API Version: 11.2, Runtime API Version: 11.2
W0321 13:57:01.620791 73276 gpu_resources.cc:91] device: 0, cuDNN Version: 8.1.
W0321 13:57:01.623584 73276 gpu_resources.cc:217] WARNING: device: . The installed Paddle is compiled with CUDNN 8.2, but CUDNN version in your machine is 8.1, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
Tensor(shape=[2, 250080], dtype=float32, place=Place(gpu:0), stop_gradient=True,
       [[0.00000071, 0.00000376, 0.00000207, ..., 0.00000050, 0.00000634,
         0.00000307],
        [0.00000094, 0.00000363, 0.00000156, ..., 0.00000039, 0.00000460,
         0.00000369]])

# PyTorch 版本
>>> torch.nn.functional.softmax(torch.tensor(x.numpy()), dim=-1)
tensor([[7.1335e-07, 3.7550e-06, 2.0744e-06,  ..., 5.0486e-07, 6.3362e-06,
         3.0748e-06],
        [9.3548e-07, 3.6312e-06, 1.5568e-06,  ..., 3.8749e-07, 4.6010e-06,
         3.6902e-06]])
```

**Comment by rajveer43** [comment]
Would like to work on this, can you give more insights?

**Comment by smallpoxscattered** [comment]
想了解一下
